Library/Homebrew/cask/dsl/caveats.rb

```diff
@@ -114,15 +114,16 @@ def eval_caveats(&block)
             #{@cask} requires Java. You can install the latest version with:
               brew install --cask adoptopenjdk
           EOS
-        elsif java_version.include?("11") || java_version.include?("+")
+        elsif java_version.include?("+")
           <<~EOS
             #{@cask} requires Java #{java_version}. You can install the latest version with:
               brew install --cask adoptopenjdk
           EOS
         else
           <<~EOS
             #{@cask} requires Java #{java_version}. You can install it with:
-              brew install --cask homebrew/cask-versions/adoptopenjdk#{java_version}
+              brew tap adoptopenjdk/openjdk
```

Comment:
We don't want to recommend third-party taps not part of the Homebrew organisation for security reasons.

Reply:
> In [Homebrew/homebrew-cask#54074](https://github.com/Homebrew/homebrew-cask/pull/54074), @claui, @vitorgalvao, and @commitay discussed this in some detail and approved this approach.

There is a huge difference between saying “we don’t support other versions of this software; if you want them you can get check this external tap” (what [Homebrew/homebrew-cask#54074](https://github.com/Homebrew/homebrew-cask/pull/54074) does) and “to install this piece of software you *have to* install software from a third-party tap” (what this does).

The former gives you a choice on the tool itself—you have to understand what it is. The latter is a requirement of a dependency—you may not understand the implication, only that it’s stopping you from getting something else.

I agree with @MikeMcQuaid, with one clarification:

> We don't want to ~~recommend~~ require third-party taps not part of the Homebrew organisation for security reasons.

Reply:
Agreed. That said: I don't think we should recommend or require them (outside of https://docs.brew.sh/Interesting-Taps-and-Forks) because your typical user will just do what the instructions say to fix their problem which puts an onus on us to ensure that their security meets our standards (which we cannot do).

Reply:
I’m fine with [removing the recommendation from the other cask](https://github.com/Homebrew/homebrew-cask/pull/104883).

Reply:
@vitorgalvao I understand the concern. From the perspective of a maintainer of a cask that *currently* is not yet confirmed to work with the version of Java that the Homebrew organization offers, I would be in favor of a solution that allows cask maintainers to advise users how to safely use the cask. We are actively working on ensuring compatibility with the current version of Java (16), but in the meantime, a "caveat" seems like the right medicine (and certainly better than nothing). Perhaps rather than endorsing a third-party tap, this language could be used:

> We don't support other versions of this dependency. Consult the project's documentation to locate a third-party tap or source for the dependency.

Would that satisfy the concern about not endorsing third-party taps for security reasons?

Reply:
> From the perspective of a maintainer of a cask that _currently_ is not yet confirmed to work with the version of Java that the Homebrew organization offers

Is it known to be broken or just not confirmed to work? 

> We don't support other versions of this dependency.

So: we don't support any of the "official" Homebrew tap versions at all? In that case I'd say this cask should probably be removed from homebrew-cask entirely.

Reply:
Ah, I see the official `homebrew/cask-versions` contains `adoptopenjdk8`, which the `exist-db` cask is certainly compatible with. 

Given the direction that this discussion has taken—i.e. toward removing mentions of third party casks—I will submit a PR updating the `exist-db` cask's caveat to remove mention of `"11"`, to read: `depends_on_java: "8"`.